datawig/simple_imputer.py

```diff
@@ -466,6 +486,62 @@ def predict(self,
 
         return imputations
 
+    @staticmethod
+    def complete(data_frame: pd.DataFrame,
+                 precision_threshold: float = 0.0,
+                 inplace: bool = False):
+        """
+        Given a dataframe with missing values, this function detects all imputable columns, trains an imputation model
+        on all other columns and imputes values for each missing value.
+
+        Imputable columns are either numeric columns or non-numeric categorical columns; for determining whether a
+            column is categorical (as opposed to a plain text column) we use the following heuristic:
+            a non-numeric categorical column should have least 10 times as many rows as there were unique values
+
+        If an imputation model did not reach the precision specified in the precision_threshold parameter for a given
+            imputation value, that value will not be imputed; thus depending on the precision_threshold, the returned
+            dataframe can still contain some missing values.
+
+        For numeric columns, we do not filter for accuracy.
+
+        :param data_frame: original dataframe
+        :param precision_threshold: precision threshold for categorical imputations (default: 0.0)
+        :param inplace: whether or not to perform imputations inplace
+        :return: dataframe with imputations
+
+        """
+
+        # TODO: should we expose temporary dir for model serialization to avoid crashes due to not-writable dirs?
+
```

Comment:
Yes. We could also allow the user to not write to disk. I've been having trouble with low disk space.

Reply:
ah that's actually a super old TODO, not part of this review. 
i guess the low disk space issue came up with count/tfidf vectorizers and/or hpo? in that case i guess, as there is a ton of boilerplate code already around the actual Imputer invocations, one might as well remove the dir as part of the hpo. 

The TODO was really just about avoiding crashes due to permissions. The consensus of the team was to not have the serialization as an option, but to always serialize.

Reply:
I create an issue for this to close here.